Makefile

```diff
@@ -73,7 +74,9 @@ $(EXAMPLE_DIRS): lib
 examples: $(EXAMPLE_DIRS)
 	$(Q)true
 
-clean: $(EXAMPLE_DIRS:=.clean) styleclean
+examplesclean: $(EXAMPLE_DIRS:=.clean) styleclean
+
+clean: examplesclean
 	$(Q)$(MAKE) -C libopencm3 clean
```

Comment:
cascade is intentional.  It's only really split to allow people (me) to leave the library built, but run clean and rebuild on the examples only while testing makefile changes in the examples.

But yeah, relooking at it, I guess it should really be:
```
examplesclean: $(EXAMPLE_DIRS:=.clean)
clean: examplesclean styleclean
styleclean: #existing styleclean stuff
```

So it's more of a tree instead of a linear order right? Is that what you were getting at?